line_obfuscator.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"golang.org/x/tools/go/ast/astutil"
+	mathrand "math/rand"
+)
+
+// PosMax is the largest line or column value that can be represented without loss.
+// Incoming values (arguments) larger than PosMax will be set to PosMax.
+// Source: https://golang.org/src/cmd/compile/internal/syntax/pos.go
+const PosMax = 1 << 30
+const PosMin = 1
+
+var emptyLine = &ast.CommentGroup{List: []*ast.Comment{{Text: "//line :1"}}}
+
+func transformLineInfo(fileIndex int, file *ast.File) *ast.File {
+	pre := func(cursor *astutil.Cursor) bool {
+		funcDecl, ok := cursor.Node().(*ast.FuncDecl)
+		if !ok {
+			return true
+		}
+
+		// Ignore functions with //go: directives
+		if funcDecl.Doc != nil && len(funcDecl.Doc.List) != 0 {
```

Comment:
Isn't this just ignoring all functions with comments?

Reply:
Yeah. I relied on clearing the comments in transformGo, but it's not called up at this stage.

Reply:
Why are you ignoring functions with `//go: directives?`

Reply:
I am working on how to add comments to all functions. Ast behaves very strangely when adding comments to already existing comments.